extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/WebSocketEndpointDecorator.java

```diff
@@ -0,0 +1,143 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import java.util.List;
+import java.util.Map;
+
+import io.quarkus.websockets.next.WebSocketClientConnection;
+import io.quarkus.websockets.next.WebSocketConnection;
+import io.quarkus.websockets.next.runtime.WebSocketEndpoint;
+
+/**
+ * Decorates {@link WebSocketEndpoint} and optionally {@link SendingInterceptor}.
+ */
+sealed interface WebSocketEndpointDecorator permits OtelWebSocketEndpointDecorator, MicrometerWebSocketEndpointDecorator {
```

Comment:
I know that the "decorator" in the name here is just a reference to the pattern but given the fact that the implementations are CDI beans it's a little bit confusing (because of CDI decorators)...

Reply:
As of now, no CDI bean has "decorator" in name. It took quite a lot of refactoring as I couldn't come with a better name.